lib/Controller/DraftsController.php

```diff
@@ -0,0 +1,154 @@
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http;
+use OCP\IRequest;
+
+class DraftsController extends Controller {
+	private DraftsService $service;
+	private string $userId;
+	private AccountService $accountService;
+
+	public function __construct(string $appName,
+								$UserId,
+								IRequest $request,
+								DraftsService $service,
+								AccountService $accountService) {
+		parent::__construct($appName, $request);
+		$this->userId = $UserId;
+		$this->service = $service;
+		$this->accountService = $accountService;
+	}
+
+	/**
+	 * @NoAdminRequired
+	 * @TrapError
+	 *
+	 * @param int $accountId
+	 * @param string $subject
+	 * @param string $body
+	 * @param string $editorBody
+	 * @param bool $isHtml
+	 * @param array $to i. e. [['label' => 'Linus', 'email' => 'tent@stardewvalley.com'], ['label' => 'Pierre', 'email' => 'generalstore@stardewvalley.com']]
+	 * @param array $cc
+	 * @param array $bcc
+	 * @param array $attachments
+	 * @param int|null $aliasId
+	 * @param string|null $inReplyToMessageId
+	 * @return JsonResponse
+	 * @throws ClientException
+	 */
+	public function create(
+		int     $accountId,
+		string  $subject,
+		string  $body,
+		string  $editorBody,
+		bool    $isHtml,
+		array   $to = [],
+		array   $cc = [],
+		array   $bcc = [],
+		array   $attachments = [],
+		?int    $aliasId = null,
+		?string $inReplyToMessageId = null) : JsonResponse {
+		$account = $this->accountService->find($this->userId, $accountId);
+
+		$message = new LocalMessage();
+		$message->setType(LocalMessage::TYPE_DRAFT);
+		$message->setAccountId($accountId);
+		$message->setAliasId($aliasId);
+		$message->setSubject($subject);
+		$message->setBody($body);
+		$message->setEditorBody($editorBody);
+		$message->setHtml($isHtml);
+		$message->setInReplyToMessageId($inReplyToMessageId);
+		$message->setSendAt(null);
+
+		$this->service->saveMessage($account, $message, $to, $cc, $bcc, $attachments);
+
+		return JsonResponse::success($message, Http::STATUS_CREATED);
+	}
+
+
+	/**
+	 * @NoAdminRequired
+	 * @TrapError
+	 *
+	 * @param int $id
+	 * @param int $accountId
+	 * @param string $subject
+	 * @param string $body
+	 * @param string $editorBody
+	 * @param bool $isHtml
+	 * @param bool $failed
+	 * @param array $to i. e. [['label' => 'Linus', 'email' => 'tent@stardewvalley.com'], ['label' => 'Pierre', 'email' => 'generalstore@stardewvalley.com']]
+	 * @param array $cc
+	 * @param array $bcc
+	 * @param array $attachments
+	 * @param int|null $aliasId
+	 * @param string|null $inReplyToMessageId
+	 * @param int|null $sendAt
+	 * @return JsonResponse
+	 */
+	public function update(int     $id,
+		int     $accountId,
+		string  $subject,
+		string  $body,
+		string  $editorBody,
+		bool    $isHtml,
+		bool    $failed = false,
+		array   $to = [],
+		array   $cc = [],
+		array   $bcc = [],
+		array   $attachments = [],
+		?int    $aliasId = null,
+		?string $inReplyToMessageId = null,
+		?int $sendAt = null): JsonResponse {
```

Comment:
I don't think this should ever not be not for a draft message, should it? :thinking:

Reply:
Unless the update method also handles conversion to an outbox message message

Reply:
I think it would be good to have that in the update. Use case creating and continuosly updating a draft and then sending it comes to mind.

Reply:
So it is the conversion aspect

Reply:
Current logic: if there's a sendAt timestamp, the message will be converted to an outbox message.

Reply:
Am I ok to resolve?